File: src/main/java/io/awilson/timein/domain/Course.java
package io.awilson.timein.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@ToString(exclude = {"students"})
@EqualsAndHashCode(exclude = {"students"})
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    @Version
    private Integer version;
    private String title;
    @OneToMany(mappedBy = "course")
    private List<Student> students;
}

File: src/main/java/io/awilson/timein/domain/Course.kt
package io.awilson.timein.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import java.util.*
import javax.persistence.*

@Entity
class Course {
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Int = 0
    @Version
    var version: Int = 0
    var title: String = ""
    @OneToMany(mappedBy = "course")
    var students: MutableList<Student> = ArrayList()
}
